cmd/login/wc.go

```diff
@@ -90,52 +61,134 @@ func (r *runner) createClusterClientCert(ctx context.Context, client k8sclient.I
 		}
 		releaseService, err = release.New(serviceConfig)
 		if err != nil {
-			return microerror.Mask(err)
+			return serviceSet{}, microerror.Mask(err)
 		}
 	}
 
+	return serviceSet{
+		clientCertService:   clientCertService,
+		organizationService: organizationService,
+		clusterService:      clusterService,
+		releaseService:      releaseService,
+	}, nil
+}
+
+func (r *runner) getCluster(ctx context.Context, services serviceSet, provider string) (*cluster.Cluster, error) {
+	var err error
 	var c *cluster.Cluster
-	{
-		var namespaces []string
-		if len(r.flag.WCOrganization) > 0 {
-			orgNamespace, err := getOrganizationNamespace(ctx, organizationService, r.flag.WCOrganization)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-
-			namespaces = append(namespaces, orgNamespace)
-		} else {
-			namespaces, err = getAllOrganizationNamespaces(ctx, organizationService)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-		}
 
-		if r.flag.WCInsecureNamespace {
-			namespaces = append(namespaces, "default")
+	var namespaces []string
+	if len(r.flag.WCOrganization) > 0 {
+		orgNamespace, err := getOrganizationNamespace(ctx, services.organizationService, r.flag.WCOrganization)
+		if err != nil {
+			return nil, microerror.Mask(err)
 		}
 
-		c, err = findCluster(ctx, clusterService, organizationService, provider, r.flag.WCName, namespaces...)
+		namespaces = append(namespaces, orgNamespace)
+	} else {
+		namespaces, err = getAllOrganizationNamespaces(ctx, services.organizationService)
 		if err != nil {
-			return microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
 	}
 
+	if r.flag.WCInsecureNamespace {
+		namespaces = append(namespaces, "default")
+	}
+
+	c, err = findCluster(ctx, services.clusterService, services.organizationService, provider, r.flag.WCName, namespaces...)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	return c, nil
+}
+
+func (r *runner) getCertOperatorVersion(c *cluster.Cluster, provider string, services serviceSet, ctx context.Context) (string, error) {
```

Comment:
Moving logic to get cert operator version into its own function. In openstack this is not needed and therefore empty.